### PR TITLE
#8106 Clarify JS init component key

### PR DESCRIPTION
**Pull request by DenisSaltanahmedov**
## Purpose of this pull request
This pull request adds additional examples of the "component key" related to JS data-mage-init to this topic.

## Affected DevDocs pages
-  https://devdocs.magento.com/guides/v2.4/javascript-dev-guide/javascript/js_init.html

## Links to Magento source code
-  src/guides/v2.4/javascript-dev-guide/javascript/js_init.md

### CODE REVIEW

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 1]
```diff
@@ -111,6 +111,20 @@ On DOM ready, the `data-mage-init` attribute is parsed to extract component name
   };
   ```
 
+Where `<component_name>` native Magento JS component, for example: `menu`, `collapsible`, `tooltip` ...
+
+```html
+<nav data-mage-init='{"tooltip": {"content": "<?= /* @noEscape */ $content ?>"}}'></nav>
+```
+
+Or a custom JS component, can be component path `Vendor_Module/js/component` or alias declareted in `requirejs-config.js`.
```
Comment (BarnyShergold) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:120]:
```suggestion
Or a custom JS component, can be component path `Vendor_Module/js/component` or alias declared in `requirejs-config.js`.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Just one small correction

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 2]
```diff
@@ -111,6 +111,20 @@ On DOM ready, the `data-mage-init` attribute is parsed to extract component name
   };
   ```
 
+Where `<component_name>` native Magento JS component, for example: `menu`, `collapsible`, `tooltip` ...
```
Comment (dobooth) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:114]:
```suggestion
Where `<component_name>` is a native Magento JS component, for example: `menu`, `collapsible`, `tooltip` ...
```

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md (hunk 1 above)
Comment (dobooth) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:120]:
```suggestion
Or a custom JS component, implemented with a component path: `Vendor_Module/js/component`, or as an alias declared in `requirejs-config.js`.
```
If this suggestion is not accurate, than some kind of grammar fix is needed here.

File: src/guides/v2.3/javascript-dev-guide/javascript/js_init.md [hunk 3]
```diff
@@ -111,6 +111,20 @@ On DOM ready, the `data-mage-init` attribute is parsed to extract component name
...
+<nav data-mage-init='{"tooltip": {"content": "<?= /* @noEscape */ $content ?>"}}'></nav>
+```
+
+Or a custom JS component, can be component path `Vendor_Module/js/component` or alias declareted in `requirejs-config.js`.
+
+```html
+<nav data-mage-init='{"Vendor_Module/js/component": {"status":"<?= /* @noEscape */ $block->getStatus(); ?>"}}'></nav>
+```
+
+More about [locate JS components](https://devdocs.magento.com/guides/v2.4/javascript-dev-guide/javascript/js_debug.html).
```
Comment (dobooth) [inline src/guides/v2.3/javascript-dev-guide/javascript/js_init.md:126]:
```suggestion
Read more about [locate JS components](https://devdocs.magento.com/guides/v2.4/javascript-dev-guide/javascript/js_debug.html).
```

**Review by dobooth — CHANGES_REQUESTED** [review]

**Comment by DenisSaltanahmedov** [comment]
@BarnyShergold, @dobooth thanks for your help. I have fixed the review.

**Review by BarnyShergold — APPROVED** [review]

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**